src/main/scala/util/PlusArg.scala

```diff
@@ -0,0 +1,20 @@
+// See LICENSE.SiFive for license details.
+
+package util
+import Chisel._
+
+class plusarg_reader(format: String, default: Int) extends BlackBox(Map(
+    "FORMAT"  -> chisel3.core.StringParam(format),
+    "DEFAULT" -> chisel3.core.IntParam(default))) {
+  val io = new Bundle {
+    val out = UInt(OUTPUT, width = 32)
+  }
+}
+
+object PlusArg
+{
+  // PlusArg("foo") will return 42 if the simulation is run with +foo=42
+  // Do not use this as an initial register value.
```

Comment:
Comments should explain why not (it is set in an initial block which would cause a race condition if used in any other initial block).